crm_lead_followers/views/crm.xml

```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+  <record id="crm_lead_view_search" model="ir.ui.view">
+    <field name="name">lead search view: add shared leads to my opportunities</field>
+    <field name="model">crm.lead</field>
+    <field name="inherit_id" ref="crm.view_crm_case_opportunities_filter"/>
+    <field name="arch" type="xml">
+      <filter name="assigned_to_me" position="attributes">
+        <attribute name="domain">['|', ('user_id', '=', uid), ('message_is_follower', '=', True)]</attribute>
```

Comment:
I think that the "assigned to me" filter should not include every lead you follow. That would be misleading.
It's better to either add a new filter or just add nothing, since anyways permissions will filter out those that you don't follow (and thus have no access to).